src/coreclr/src/jit/compiler.cpp

```diff
@@ -512,6 +512,55 @@ bool Compiler::isSingleFloat32Struct(CORINFO_CLASS_HANDLE clsHnd)
 }
 #endif // ARM_SOFTFP
 
+#ifdef TARGET_X86
+//---------------------------------------------------------------------------
+// isTrivialPointerSizedStruct:
+//    Check if the given struct type contains only one pointer-sized integer value type
+//
+// Arguments:
+//    clsHnd - the handle for the struct type.
+//
+// Return Value:
+//    true if the given struct type contains only one pointer-sized integer value type,
+//    false otherwise.
+//
+bool Compiler::isTrivialPointerSizedStruct(CORINFO_CLASS_HANDLE clsHnd) const
+{
+    assert(info.compCompHnd->isValueClass(clsHnd));
+    if (info.compCompHnd->getClassSize(clsHnd) != TARGET_POINTER_SIZE)
+    {
+        return false;
+    }
+    for (;;)
```

Comment:
This is a lot of back and forth with the EE; would it make sense to do all this work over on the jit interface side?

(I'm guessing we rarely iterate in this loop, but still...)

Reply:
It's very unlikely we'll iterate through this loop more than once in my experience, but I can move this to the other side of the ee-jit interface if you think that's better for perf. I was trying to avoid the hassle of versioning the jit-ee interface, but I'm fine doing it.

Reply:
I'd be fine if you just measured how often we have to iterate (over say PMI FX) and just confirmed it is truly rare.

Reply:
I've run the PMI FX diff as you recommended and confirmed that in the whole framework we run the loop body at most twice.

We run the loop body twice in 557 cases (551 cases are in Roslyn, the remaining 6 are in System.Reflection.Metadata), and we run the loop body only once in 102592 cases.

So, we run the loop body only once in 0.5% of cases if we include the Roslyn assemblies in Core_Root, or approximately 0.005% of cases if we exclude the Roslyn assemblies.

Reply:
Also, we only make it past the first checks in the function (the pointer-sized check and the first "is value type with one field" check) 19% of the time.